lightning/src/chain/transaction.rs

```diff
@@ -73,6 +73,14 @@ impl OutPoint {
 			vout: self.index as u32,
 		}
 	}
+
+	/// Creates a dummy BitcoinOutPoint, useful for serializing into.
+	pub(crate) fn null() -> Self {
+		Self {
+			txid: Default::default(),
+			index: 0
+		}
+	}
```

Comment:
Yea, so initially I went down a rabbit hole on creating a whole trait for this stuff but I eventually just gave up and created a wrapper that deserializes into `Option`s. This one is left because its used outside of the `impl_writeable_tlv_based` macro, so its kinda a one-off.

Reply:
If a one-off, consider inlining.

Reply:
I found it to be a bit cleaner, but I'm happy to inline in channelmanager.rs if you prefer?